components/kyma-environment-broker/internal/suspension/handler.go

```diff
@@ -84,6 +84,10 @@ func (h *ContextUpdateHandler) handleContextChange(newCtx internal.ERSContext, i
 	}
 
 	if *newCtx.Active {
+		if lastDeprovisioning != nil && !lastDeprovisioning.Temporary {
```

Comment:
I think skipping the unsuspension event is not correct. It is a valid scenario to receive unsuspension while suspension is still in progress. KEB should queue the unsuspension operation but keep it pending until the suspension is completed.